### DISCUSSION TITLE
Fundamentos do ASP.NET 6 - Associando Role com Usuários

### DISCUSSION THREAD

**Discussion by mgpaixao** [balta-io/2811#4] [Q&A] [answered]
Balta, no curso Fundamentos do ASP.NET 6 no video **_Gerando tokens com Claims_**, você faz uma associação no próprio banco de dados do User com o Role dele certo?

Porém, quando eu pego o codigo pronto e rodo, na criação de User, não se associa um Role com um User, então nada é salvo na tabela intermediaria UserRole.

Como posso fazer para que se associe esse Role "básico" à um User logo no registro/crianção de um novo Usuário através do endpoint Post da rota "/accounts" ?

Eu não quero fazer essa associação na mão, eu quero que quando eu registre um novo usuário ele já saia com um role "basico" já associado e registrado na tabela UserRole

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @mgpaixao, como vai?

Para fazer esta associação basta adicionar um role ao `User` durante o cadastro.
Ficaria algo assim:

```csharp
var role = context.Roles.FirstOrDefault(x => x.Name == "admin");
user.Roles.Add(role);

context.Users.Add(user);
context.SaveChanges();
```
